.evergreen/compile-unix.sh

```diff
@@ -65,6 +65,11 @@ esac
 # Report the current PHP version
 echo "PHP: `php --version | head -n 1`"
 
+# If we're testing a specific version of libmongoc, update submodule sources
+if [ -n "$LIBMONGOC_VERSION" ]; then
+   php scripts/update-submodule-sources.php
+fi
```

Comment:
Below we have a conditional for `make libmongoc-version-current`. I suppose that still needs to remain separate because `phpize` is what builds `Makefile` from `Makefile.frag`. So, at best, we could move that _between_ the `phpize` and `./configure` lines below.

And on a related note: the `LIBMONGOC_VERSION_CURRENT` file is actually referenced _during_ `./configure` to create the substitutions, so shouldn't we be running `make libmongoc-version-current` _before_ `./configure`? As-is, I don't think the compiled driver reports the correct version info.

Do you want to handle that in this same PR?

Reply:
Good point: `LIBMONGOC_VERSION_CURRENT` is indeed referenced during `./configure`, but we have to run `./configure` in order to have the `Makefile` generated. To avoid that conundrum, I now call `phpize` and `./configure`, then run `make libmongoc-version-current` before running `./configure` again to generate the final `Makefile` that we're building with. It's a little messy but the only way to work around the issue.

Reply:
I wouldn't complain if you want to just manually run:

```
cd src/libmongoc/ && python build/calc_release_version.py > ../LIBMONGOC_VERSION_CURRENT
```